### ISSUE TITLE
Flicker that causes loss of focus on input when clicking on next input

### ISSUE THREAD

**Issue by npyl** [pdfme/pdfme#492]
### Describe the bug

Hello guys, I have a React 18, NextJS 13 application.
I am using the Form class to render a pdf file along with inputs that correspond to a schema I have generated.
My problem is that if I type something on a random input, and then click on another, I will get a flicker and this flicker will cause loss of focus, meaning I have to click again. Most of the times, the second click isn't enough and I click 3 times just to be sure.

This is a very big problem for me because I have a form that contains approximately 20 input fields.

### To Reproduce

1. Type on an input
2. Click on another input
You will see a re-render of the inputs
3. Click 1-2 more times to be able to type.

### Expected behavior

1. I type on an input
2. I click on another
3. It is expected to be able to type something straight away.

### Your Environment

```markdown
- pdfme package(@pdfme/generator or @pdfme/ui):@pdfme/ui
- pdfme version: 4.0.2
- Operating system: Windows 10
- Node.js version or Browser name & version: Node 20.12.2, Firefox, Chrome
```

### Your Error Log

```shell
Warning: Setting up fake worker.

** THE FOLLOWING HAPPENS every time a re-render happens for every field **
index.es.js:12091 Warning: TT: undefined function: 32
```

**Comment by npyl** [comment]
I should also note that this does not happen with 3.4.3 version.

**Comment by peteward** [comment]
Confirmed in the Form on the playground site using Certificate.

It only takes me one additional click though.

It looks like it's processing the change event on loss of focus and not focusing the new element.

**Comment by npyl** [comment]
Yes my guess was that it was trying to process the content of the input in case the user has written something.

**Comment by peteward** [comment]
This is due to a known issue [here](https://github.com/pdfme/pdfme/blob/main/packages/ui/src/components/Preview.tsx#L141-L154).

We're triggering a full re-render of the Form because a table would require this but it's applying to all other field types as well.

A clean design is needed to handle this, although there might be a short-term improvement that we could make.

**Comment by peteward** [comment]
This should be fixed now in the next release

**Comment by npyl** [comment]
Thanks alot!!

**Comment by hand-dot** [comment]
Hi @npyl 
Released in version 4.1.1.
If the issue has been resolved in the latest version, please close this issue.
